src/plugins/es_ui_shared/__packages_do_not_import__/flyout_multi_content/flyout_multi_content.tsx

```diff
@@ -0,0 +1,170 @@
+/*
```

Comment:
So this is a hard one. 😊 

I am not sure about the `Singleton` name but what about `GlobalFlyout`.
So the component that consumes it would have

```
const flyout = useGlobalFlyout();

flyout.addContent(...);
```

Another option could be `AppFlyout`, as there should only be 1 per app

```
const flyout = useAppFlyout();

flyout.addContent(...);
```

WDYT?

Reply:
I think `GlobalFlyout` is pretty good. In fact, this could perhaps even be provided by EUI at some point in the future! Or at least a recommended pattern.

Reply:
Yeah I thought about it too, once it will have a README + test we will ask if EUI wants to provide it themselves.